main.cc

```diff
@@ -170,6 +171,54 @@ read_config(bpo::variables_map& opts, db::config& cfg) {
     });
 }
 
+struct object_storage_endpoint_param {
+    sstring endpoint;
+    s3::endpoint_config config;
+};
+
+namespace YAML {
+template<>
+struct convert<::object_storage_endpoint_param> {
+    static bool decode(const Node& node, ::object_storage_endpoint_param& ep) {
+        ep.endpoint = node["name"].as<std::string>();
+        return true;
+    }
+};
+}
+
+// run inside async context
+static std::unordered_map<sstring, s3::endpoint_config> read_object_storage_config(const db::config& cfg) {
+    std::unordered_map<sstring, s3::endpoint_config> ret;
+    sstring file;
+    if (!cfg.object_storage_config_file().empty()) {
+        file = cfg.object_storage_config_file();
+    } else {
+        file = db::config::get_conf_sub("object_storage.yaml").native();
+    }
+    YAML::Node doc;
+    try {
+        doc = YAML::LoadFile(file);
```

Comment:
Will this stall?

Reply:
It can, yes. On boot, which's probably not that bad. But since we're going to make the file re-readable on HUP, then it needs to be fixed.

Reply:
Fixed